### PR TITLE
PLAT-1273: Add trigger events

### PR DESCRIPTION
**Pull request by mihaelpalfi**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-1273)

#### Description
As per JIRA ticket

- Also checked that events are fired only once.
- Added also check that on hover when mouse is moving it dosent execute the show popper a milion times.

#### ChangeLOG

[Changelog](https://zen-ui.zengrc.com/?path=/docs/changelog--page)

#### Useful links
[Zen UI readme](https://github.com/reciprocity/zen-ui/blob/main/README.md)

[PR review guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/pull_request_reviews.md)

[Git commit guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/commit_guidelines.md)

### CODE REVIEW

File: src/components/zen-popover/zen-popover.stories.mdx [hunk 1]
```diff
@@ -179,21 +211,29 @@ Popover component
...
 
-Show popover bottom start and bottom end.
+Show popover position examples. ()
 
 <Canvas>
   <Story name="StoryPositionVariants">{StoryPositionVariants.bind({})}</Story>
 </Canvas>
 
-### 2. Story with positioning set to top
+### 3. Story different trigger events
```
Comment (chavlji) [inline src/components/zen-popover/zen-popover.stories.mdx:220]: Leave `Story` out in all of these titles. So in this case keep just `Different trigger events`.
(sorry I haven't understood the question yesterday)

File: src/components/zen-popover/zen-popover.stories.mdx [hunk 2]
```diff
@@ -179,21 +211,29 @@ Popover component
 
 ### 1. Story position variants
 
-Show popover bottom start and bottom end.
+Show popover position examples. ()
```
Comment (chavlji) [inline src/components/zen-popover/zen-popover.stories.mdx:214]: Remove `()`

Comment (chavlji) [reply src/components/zen-popover/zen-popover.stories.mdx:214]: Story description should try to describe how to achieve the shown behaviour:
`Use attribute placement to set default side of trigger element on which popover will be showed`
Apply please to all stories in this file.

File: src/components/zen-popover/zen-popover.tsx [hunk 3]
```diff
@@ -20,6 +21,12 @@ export class ZenPopover {
   /** Placement */
   @Prop() readonly placement: Placement = 'bottom-end';
 
+  /** Trigger event */
+  @Prop() readonly triggerEvent: TriggerEvent = 'click';
```
Comment (chavlji) [inline src/components/zen-popover/zen-popover.tsx:25]: We're mixing words `trigger` and `target`. Decide for one and use it in all cases within popover component.

Comment (chavlji) [reply src/components/zen-popover/zen-popover.tsx:25]: `triggeringEvent` would be better.

File: src/components/zen-popover/zen-popover.tsx [hunk 4]
```diff
@@ -20,6 +21,12 @@ export class ZenPopover {
...
+
+  /** Dont hide tooltip */
```
Comment (chavlji) [inline src/components/zen-popover/zen-popover.tsx:27]: Don't

File: src/components/zen-popover/zen-popover.tsx [hunk 5]
```diff
@@ -31,73 +38,103 @@ export class ZenPopover {
...
+      this.addTriggerEvents();
+      this.hide();
+    }
   }
 
-  destroyPopper() {
-    if (this.popperInstance) {
-      this.popperInstance.destroy();
-      this.popperInstance = null;
+  addTriggerEvents() {
```
Comment (chavlji) [inline src/components/zen-popover/zen-popover.tsx:66]: lint error: missing return type (void)
Please fix for all functions in this file.

File: src/components/zen-popover/zen-popover.tsx [hunk 6]
```diff
@@ -20,6 +21,12 @@ export class ZenPopover {
   /** Placement */
   @Prop() readonly placement: Placement = 'bottom-end';
```
Comment (chavlji) [inline src/components/zen-popover/zen-popover.tsx:22]: Please rename to `position` to be the same as in `tooltip`

**Review by chavlji — CHANGES_REQUESTED** [review]
A few minor corrections, but it works as expected, and code in general looks clean and readable!

File: src/components/zen-popover/zen-popover.stories.mdx (hunk 1 above)
Comment (mihaelpalfi) [reply src/components/zen-popover/zen-popover.stories.mdx:220]: Changed

File: src/components/zen-popover/zen-popover.tsx (hunk 3 above)
Comment (mihaelpalfi) [reply src/components/zen-popover/zen-popover.tsx:25]: Dont quite understand. In my opinion TriggerEvent or TargetEvent event would be ok. I can add triggeringEvent to the description of param maybe?

File: src/components/zen-popover/zen-popover.tsx (hunk 4 above)
Comment (mihaelpalfi) [reply src/components/zen-popover/zen-popover.tsx:27]: Done

File: src/components/zen-popover/zen-popover.tsx (hunk 5 above)
Comment (mihaelpalfi) [reply src/components/zen-popover/zen-popover.tsx:66]: Added. Dont get lint errors for this.

File: src/components/zen-popover/zen-popover.tsx (hunk 6 above)
Comment (mihaelpalfi) [reply src/components/zen-popover/zen-popover.tsx:22]: Done.

File: src/components/zen-popover/zen-popover.stories.mdx (hunk 2 above)
Comment (chavlji) [reply src/components/zen-popover/zen-popover.stories.mdx]: This one hasn't been done.

File: src/components/zen-popover/zen-popover.tsx (hunk 3 above)
Comment (chavlji) [reply src/components/zen-popover/zen-popover.tsx:25]: Ah ok, my bad. Trigger always refers to the event and not to the target element.

File: src/components/zen-popover/zen-popover.tsx (hunk 5 above)
Comment (chavlji) [reply src/components/zen-popover/zen-popover.tsx]: Hm strange. I do get them in vscode.

**Review by chavlji — CHANGES_REQUESTED** [review]
Fix story descriptions.

File: src/components/zen-popover/zen-popover.stories.mdx (hunk 2 above)
Comment (mihaelpalfi) [reply src/components/zen-popover/zen-popover.stories.mdx]: Sorry missed this one. Done.

**Review by chavlji — APPROVED** [review]

**Pull request merged**